src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -89,9 +89,27 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
 
         protected abstract Task<AuthenticateResult> HandleRemoteAuthenticateAsync();
 
-        protected override Task<AuthenticateResult> HandleAuthenticateAsync()
+        protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         {
-            return Task.FromResult(AuthenticateResult.Fail("Remote authentication does not support authenticate"));
+            var authenticateContext = new AuthenticateContext(Options.SignInScheme);
+            await PriorHandler.AuthenticateAsync(authenticateContext);
+            if (authenticateContext.Accepted)
+            {
+                if (authenticateContext.Error != null)
+                {
+                    return AuthenticateResult.Fail(authenticateContext.Error);
+                }
+                // Compare the ClaimsIssuer to make sure the identity was produced by this provider and not by some other one.
+                if (string.Equals(authenticateContext.Principal?.Identity?.AuthenticationType, Options.ClaimsIssuer, StringComparison.Ordinal))
```

Comment:
Also your example of an authorization policy is a bit suspect.  In general you would have a policy asking for certain claims, and then specify which auth schemes will be queried for the identities/claims.  So it's unlikely you would have a "Github" policy, you might have a "SocialAuth" policy which listed "Github" as one of the authentication schemes for the policy.  There shouldn't be anything wrong with the policy working against a claim requirement if it was satisfied from a Facebook identity or a github identity, that's sort of the point.  Its supposed to allow authorization against a set of claims, rather than worrying about where they came from.